openlibrary/plugins/openlibrary/js/onboardingCard.js

```diff
@@ -0,0 +1,100 @@
+export function initOnboardingCard() {
+    const openModal = function (e) {
+        e.preventDefault();
+        $('.card').removeClass('hidden');
+        $('.overlay').removeClass('hidden');
+    };
+
+    const closeModal = function () {
+        $('.card').addClass('hidden');
+        $('.overlay').addClass('hidden');
+    };
+
+    $('.show-modal').on('click', openModal);
+    $('.overlay').on('click', closeModal);
+
+    document.addEventListener('keydown', function (e) {
+        if (e.key === 'Escape' && !$('.modal').toggleClass('hidden')) {
+            closeModal();
+        }
+    });
+
+    const slider = function () {
+        const slides = document.querySelectorAll('.slide');
+        const btnRight = document.querySelectorAll('.card-next');
+        const dotContainer = document.querySelector('.dots');
+
+        let curSlide = 0;
+        const maxSlide = slides.length;
+        const createDots = function () {
+            slides.forEach(function (_, i) {
+                dotContainer.insertAdjacentHTML(
+                    'beforeend',
+                    `<span type="button" class='dots__dot' data-slide='${i}'></span>`
+                );
+            });
+        };
+
+        const activateDot = function (slide) {
+            document.querySelectorAll('.dots__dot').forEach((dot) => {
+                dot.classList.remove('dots__dot--active');
+            });
+            document
+                .querySelector(`.dots__dot[data-slide='${slide}']`)
+                .classList.add('dots__dot--active');
+        };
+
+        const goToSlide = function (slide) {
+            slides.forEach(
+                (s, i) => (s.style.transform = `translateX(${100 * (i - slide)}%)`)
+            );
+        };
+
+        // Next slide
+        const nextSlide = function () {
+            if (curSlide === maxSlide - 1) {
+                curSlide = 0;
+            } else {
+                curSlide++;
+            }
+
+            goToSlide(curSlide);
+            activateDot(curSlide);
+        };
+
+        const prevSlide = function () {
+            if (curSlide === 0) {
+                curSlide = maxSlide - 1;
+            } else {
+                curSlide--;
+            }
+            goToSlide(curSlide);
+            activateDot(curSlide);
+        };
+
+        const init = function () {
+            goToSlide(0);
+            createDots();
+
+            activateDot(0);
+        };
+        init();
```

Comment:
Why did you decide to put these three calls in the init function but not lines 84-98?

Reply:
The three calls are for creating the dots which are not created using html, they are created using javascript. They are put together as they are created when the dom is  loaded and script is executed.